contracts/interchain-token-service/tests/execute.rs

```diff
@@ -228,6 +229,125 @@ fn execute_hub_message_succeeds() {
     goldie::assert_json!(responses);
 }
 
+#[test]
+fn execute_message_interchain_transfer_should_scale_custom_tokens_when_decimals_are_different() {
+    let (
+        mut deps,
+        TestMessage {
+            router_message,
+            source_its_contract,
+            source_its_chain,
+            destination_its_chain,
+            destination_its_contract,
+            ..
+        },
+    ) = utils::setup_with_chain_configs(
+        Uint256::MAX.try_into().unwrap(),
+        6,
+        u64::MAX.try_into().unwrap(),
+        6,
+    );
+    let token_id = TokenId::new([1; 32]);
+    let hub_message = HubMessage::RegisterTokenMetadata(RegisterTokenMetadata {
+        decimals: 6,
+        token_address: HexBinary::from([1; 32]).try_into().unwrap(),
+    });
+    let cc_id = CrossChainId {
+        source_chain: source_its_chain.clone(),
+        message_id: router_message.cc_id.message_id.clone(),
+    };
+    assert_ok!(utils::execute_hub_message(
+        deps.as_mut(),
+        cc_id,
+        source_its_contract.clone(),
+        hub_message,
+    ));
+
+    let hub_message = HubMessage::RegisterTokenMetadata(RegisterTokenMetadata {
+        decimals: 18,
+        token_address: HexBinary::from([1; 32]).try_into().unwrap(),
+    });
+    let cc_id = CrossChainId {
+        source_chain: destination_its_chain.clone(),
+        message_id: router_message.cc_id.message_id.clone(),
+    };
+    assert_ok!(utils::execute_hub_message(
+        deps.as_mut(),
+        cc_id,
+        destination_its_contract.clone(),
+        hub_message,
+    ));
+
+    let hub_message = HubMessage::SendToHub {
+        destination_chain: destination_its_chain.clone(),
```

Comment:
include responses for RegisterTokenMetadata and LinkToken in the goldie::assert_json